pkg/utils/utils.go

```diff
@@ -56,17 +58,42 @@ func RetrieveValFromMap[K comparable, V any](m map[K]V, key K) V {
 // PollUntil validates if a certain condition is met at defined poll intervals.
 // If a timeout is reached, an associated error is returned to the caller.
 // condition contains the use-case specific code that returns true when a certain condition is achieved.
-func PollUntil(pollInterval, timeOut <-chan time.Time, condition func() (bool, error)) error {
+func PollUntil(pollInterval, timeOut <-chan time.Time, condition func() (string, bool, error)) error {
+	s := spinner.New(spinner.CharSets[11], 100*time.Millisecond)
+	s.Color("cyan")
+	startTime := time.Now()
+
+	var message string
+	var done bool
+	var err error
+	var totalElapsed time.Duration
+	secondTicker := time.NewTicker(1 * time.Second)
+	defer secondTicker.Stop()
+
+	message, done, err = condition()
+	if err != nil {
+		return fmt.Errorf("initial condition check failed: %w", err)
+	} else if done {
+		fmt.Printf("Operation completed successfully. Message: %s (Total time: %s)\n", message, totalElapsed.Round(time.Second))
+		return nil
+	}
```

Comment:
```suggestion
                       _, done, err = condition()
                        if err != nil || done {
                                return err
                        }
```
As the message will be empty when the condition is met already.

Reply:
Added this change